docs/machineregistration-reference.md

```diff
@@ -63,25 +63,28 @@ Contains the installation configuration that would be applied via `elemental-reg
+|-----------------|--------|---------------|--------------------------------------------------------------------------------------------------------------------------------------------|
+| firmware        | string | efi           | Firmware to install ('efi' or 'bios')                                                                                                      |
+| device          | string | empty         | Device to install the system to                                                                                                            |
+| device-selector | string | empty         | Rules for picking device to install the system to                                                                                          |
+| no-format       | bool   | false         | Don’t format disks. It is implied that COS_STATE, COS_RECOVERY, COS_PERSISTENT, COS_OEM partitions are already existing on the target disk |
+| config-urls     | list   | empty         | Cloud-init config files locations                                                                                                          |
+| iso             | string | empty         | Performs an installation from the ISO url instead of the running ISO                                                                       |
+| system-uri      | string | empty         | Sets the system image source and its type (e.g. 'docker:registry.org/image:tag') instead of using the running ISO                          |
+| debug           | bool   | false         | Enable debug output                                                                                                                        |
+| tty             | string | empty         | Add named tty to grub                                                                                                                      |
+| poweroff        | bool   | false         | Shutdown the system after install                                                                                                          |
+| reboot          | bool   | false         | Reboot the system after install                                                                                                            |
+| eject-cd        | bool   | false         | Try to eject the cd on reboot                                                                                                              |
 
 :::warning warning
 In case of using both `iso` and `system-uri` the `iso` value takes precedence
 :::
 
-The only required value for a successful installation is the `device` key as we need a target disk to install to. The rest of the parameters are all optional.
+It is only required to specify either the `device` or `device-selector` fields for a successful install, the rest of the parameters are all optional.
+
+If both `device` and `device-selector` is specified the value of `device` is used and `device-selector` is ignored.
```

Comment:
There should be a warning message logged for such cases.

Reply:
Good point, adding: https://github.com/rancher/elemental-operator/pull/574